core/templates/dev/head/pages/community-dashboard-page/services/contribution-opportunities-backend-api.service.ts

```diff
@@ -0,0 +1,57 @@
+// Copyright 2019 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Service for fetching the opportunities available for
+ * contributors to contribute.
+ */
+
+angular.module('oppia').factory('ContributionOpportunitiesBackendApiService', [
+  '$http', 'UrlInterpolationService', 'OPPORTUNITY_TYPE_TRANSLATION',
+  'OPPORTUNITY_TYPE_VOICEOVER', function(
+      $http, UrlInterpolationService, OPPORTUNITY_TYPE_TRANSLATION,
+      OPPORTUNITY_TYPE_VOICEOVER) {
+    return {
+      fetchTranslationOpportunities: function(
```

Comment:
I think you can extract out the common parts of these two methods into an internal function, to avoid code duplication. In general try to find ways to do that unless you expect the two cases to deviate significantly.

Reply:
No, not done. There should be a common internal function where you can just pass a single arg which is either OPPORTUNITY_TYPE_TRANSLATION or OPPORTUNITY_TYPE_VOICEOVER.